exp/addons/api/v1beta1/clusterresourcesetbinding_webhook_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestClusterResourceSetBindingClusterNameImmutable(t *testing.T) {
+	tests := []struct {
+		name           string
+		oldClusterName string
+		newClusterName string
+		expectErr      bool
+	}{
+		{
+			name:           "when the ClusterName has not changed",
+			oldClusterName: "bar",
+			newClusterName: "bar",
+			expectErr:      false,
+		},
+		{
+			name:           "when the ClusterName has changed",
+			oldClusterName: "bar",
+			newClusterName: "different",
+			expectErr:      true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			newClusterResourceSetBinding := &ClusterResourceSetBinding{
+				Spec: ClusterResourceSetBindingSpec{
+					ClusterName: tt.newClusterName,
+				},
+			}
+
+			oldClusterResourceSetBinding := &ClusterResourceSetBinding{
+				Spec: ClusterResourceSetBindingSpec{
+					ClusterName: tt.oldClusterName,
+				},
+			}
+
+			if tt.expectErr {
+				g.Expect(newClusterResourceSetBinding.ValidateUpdate(oldClusterResourceSetBinding)).NotTo(Succeed())
```

Comment:
Also run these tests on create (only with the `newClusterResourceSetBinding`. You can do it inline here, and it's the pattern we have in another few of our tests.